continuum/datasets/metashift.py

```diff
@@ -0,0 +1,186 @@
+
+        # Pickle file
+        if os.path.exists(os.path.join(self.data_path, "full-candidate-subsets.pkl")):
+            print("Classes and contexts already downloaded")
+        else:
+            file = download.download(self.pickle_url, self.data_path)
+            os.rename(file, os.path.join(self.data_path, "full-candidate-subsets.pkl"))
+            print("Classes and contexts downloaded")
+    
+    def get_data(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """Generate Metashift Data
+        Metashift classifies the Visual Genome dataset into classes and contexts. 
+        Images can be found in many classes and contexts. random_context is False, 
+        images will appear many times within the return value. We thus might get :
+
+          x       y       t
+
+        1.jpg   class1  context1
+        1.jpg   class1  context2
+        1.jpg   class2  context1
+        1.jpg   class2  context3
+        1.jpg   class2  context4
+        2.jpg   class3  context2
+        ...     ...     ...
+
+        x contains all images ids present in at least 1 context.
+        y contains the class of the image.
+        t contains the context in which the image is represented.
+        """
+        x, y, t = [], [], []
+
+        with open(os.path.join(self.data_path, "full-candidate-subsets.pkl"), 'rb') as pkl_file:
+            pkl_dict = pkl.load(pkl_file)
+        
+
+        for key in pkl_dict.keys():
+                # Iterate through all class(context) pairs
+                
+            context_name = key.split("(")[1][:-1]
+            class_name = key.split("(")[0]
+
+            if (self.class_names is None or class_name in self.class_names) :
+                
+                for id in pkl_dict[key]:    # Iterate through all ids
+
+                    if self.train_image_ids is None or id in self.train_image_ids :
+                        x.append(os.path.join(self.data_path, "images", "images", id)+".jpg")
+                        y.append(class_name)
+                        t.append(context_name)
+
+
+        x, y, t = np.array(x), np.array(y), np.array(t)
+        
+
+        if(self.unique_occurence == True):
+            x, y, t = _select_unique_occurence(x, y, t, self.seed)
+
+        self.class_names, y = np.unique(y, return_inverse = True)
+
+        if (self.strict_domain_inc == True):
+            x, y, t = _strict_domain_tasks(x, y, t, len(self.class_names))
+            t = np.unique(t, return_inverse=True)[1]
+        else:
+            t = np.unique(t, return_inverse = True)[1]
+        
+        n = np.max(t) + 1
+
+        if (self.nb_tasks > 0) and (n > self.nb_tasks): # to be tested
+            scale = lambda x : x % self.nb_tasks # scale task ids to number of tasks
+            t = scale(t)
+
+        return x, y, t
+
+def _select_unique_occurence(x, y, t, rand_seed): 
+    #choose a random context for each class(context) combiation for each image amoung available.    
```

Comment:
do you want to say that ?
`# choose an unique context for each data point (randomly)`